sudo/src/run_handler.rs

```diff
@@ -191,6 +191,21 @@ pub fn run_target(
     do_request(req, copy_env, manually_requested_dir)
 }
 
+fn search_path(filename: &str) -> Result<String> {
+    let filename = &HSTRING::from(filename);
+    let len = unsafe { SearchPathW(None, filename, None, None, None) };
+
+    if len == 0 {
+        return Err(Error::from_win32());
+    }
+
+    let mut buffer = vec![0; len as usize];
+    let len = unsafe { SearchPathW(None, filename, None, Some(&mut buffer), None) };
+    buffer.truncate(len as usize);
```

Comment:
I believe this should check the return length again here, just in case the file system contents have changed in the meantime. It could also use a loop with an eager alloc like this (pseudo-code, not tested):
```rust
fn search_path(filename: &str) -> Result<String> {
    let filename = HSTRING::from(filename);
    let mut buffer = vec![0; 260];

    loop {
        let len: usize = unsafe { SearchPathW(None, filename, None, Some(&mut buffer), None) };
    
        // If the function fails, the return value is zero.
        if len == 0 {
            return Err(Error::from_win32());
        }
    
        // If the function succeeds, the value returned is the
        // length of the string that is copied to the buffer,
        // not including the terminating null character.
          if len < buffer.size() {
              buffer.truncate(len);
              return Ok(String::from_utf16(&buffer)?);
          }
    
        // If the return value is greater than nBufferLength, the value
          // returned is the size of the buffer that is required to hold
          // the path, including the terminating null character.
          //
          // Includes some padding just in case and because it's cheap.
        buffer.resize(len + 64, 0);
    }
}
```